db/migrate/20170718111834_unify_container_definition_and_container.rb

```diff
@@ -0,0 +1,174 @@
+      end
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      port_configs = Arel::Table.new(:container_port_configs)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(port_configs[:container_definition_id])).to_sql
+      ContainerPortConfig.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      env_vars = Arel::Table.new(:container_env_vars)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(env_vars[:container_definition_id])).to_sql
+      ContainerEnvVar.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch resource_id with container_id, resource_type to 'Container' for security_contexts") do
+      security_contexts = Arel::Table.new(:security_contexts)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(security_contexts[:resource_id])
+                           .and(security_contexts[:resource_type].eq(Arel::Nodes::Quoted.new('ContainerDefinition')))).to_sql
+      SecurityContext.where(:resource_type => 'ContainerDefinition').update_all("resource_type = 'Container', resource_id = (#{join_sql})")
+    end
+
+    MiqQueue.where(:method_name => "purge_timer", :class_name => 'ContainerDefinition').destroy_all
+
+    # relationships
+    rename_column :container_port_configs, :container_definition_id, :container_id
+    rename_column :container_env_vars, :container_definition_id, :container_id
+
+    remove_column :containers, :container_definition_id
+    drop_table :container_definitions
+  end
+
+  def down
+    create_table :container_definitions do |t|
+      t.belongs_to :ems, :type => :bigint       # reconstructed columns
+      t.string     :ems_ref
+      t.bigint     :old_ems_id
+      t.timestamp  :deleted_on
+      t.string     :name
+      t.string     :image                       # copied over columns
+      t.string     :image_pull_policy
+      t.string     :memory
+      t.float      :cpu_cores
+      t.belongs_to :container_group, :type => :bigint
+      t.boolean    :privileged
+      t.bigint     :run_as_user
+      t.boolean    :run_as_non_root
+      t.string     :capabilities_add
+      t.string     :capabilities_drop
+      t.text       :command
+      t.bigint     :container_id                # temp column
+    end
+
+    add_index :container_definitions, :deleted_on
+
+    add_column :containers, :container_definition_id, :bigint
+
+    say_with_time("populate container_definitions. use container_id to keep relation to containers") do
+      insert_statement = "INSERT INTO container_definitions (container_id, ems_id, ems_ref, old_ems_id, deleted_on, name, image,
```

Comment:
Prefer this format for building SQL

```ruby
    insert_statement = <<-SQL
INSERT INTO container_defintions...
SELECT id, ...
FROM containers
SQL
```

The reason is that a) it keeps the SQL on it's own lines, and b) some IDEs will see Ruby <<-SQL block and actually color the SQL statement with SQL syntax coloring.